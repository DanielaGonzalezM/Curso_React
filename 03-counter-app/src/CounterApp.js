//rafcp
import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = (e) => {
        //setCounter(counter+1);
        setCounter((c) => c + 1);
    }

    const handleReset = (e) => setCounter(value);
    const handleSus = (e) => setCounter(counter - 1);


    return (
        <>
            <h1>CounterAPP</h1>
            <h2>{counter}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSus}>-1</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;