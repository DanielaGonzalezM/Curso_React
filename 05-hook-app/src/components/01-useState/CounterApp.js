import React, { useState } from 'react'
import './counter.css';
export const CounterApp = () => {
const [state, setState] = useState({
  counter1:10,
  counter2:20,
  counter3:30,
  counter4:40,
  counter5:50
})
const {counter1,counter2}=state;
console.log(counter1,counter2)
  return (
    <>
       <h1>Counter1 {counter1}</h1>
       <h1>Counter2 {counter2}</h1>
       <hr />
       <button 
       className='btn btn-primary' 
       onClick={()=>{
         setState({...state,counter1:counter1+1});
       }}
       >+1</button>
    </>
  )
} 
