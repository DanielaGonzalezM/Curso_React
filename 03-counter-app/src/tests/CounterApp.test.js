import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';



describe('Pruebas CounterApp', () => {
    const value = 0;
    const wrapper = shallow(<CounterApp value={value} />);

    beforeEach(()=>{  //se ejecuta antes de cada test, sirve para reiniciar las variables
        console.log('BedoreEach!');
    });

    test('debe mostrar CounterApp correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar CounterApp correctamente', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value} />);
        const h2 = wrapper.find('h2').text().trim();
        expect(parseInt(h2)).toBe(value);
    });
    test('debe aumentar con botón +1', () => {

        wrapper.find('button').at(0).simulate('click');
        const h2 = wrapper.find('h2').text().trim();
        expect(parseInt(h2)).toBe(value+1);
    });
    test('debe resetear con botón reset', () => {

        wrapper.find('button').at(1).simulate('click');
        const h2 = wrapper.find('h2').text().trim();
        expect(parseInt(h2)).toBe(value);
    });
    test('debe decrementar con botón -1', () => {

        wrapper.find('button').at(2).simulate('click');
        const h2 = wrapper.find('h2').text().trim();
        expect(parseInt(h2)).toBe(value-1);
    });

});