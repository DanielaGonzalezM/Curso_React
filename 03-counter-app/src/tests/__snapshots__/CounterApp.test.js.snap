// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas CounterApp debe mostrar CounterApp correctamente 1`] = `
<Fragment>
  <h1>
    CounterAPP
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
